keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,891 @@
+        {"input_filters": 512, "num_repeats": 4},
+    ],
+    420: [
+        {"input_filters": 64, "num_repeats": 4},
+        {"input_filters": 128, "num_repeats": 44},
+        {"input_filters": 256, "num_repeats": 87},
+        {"input_filters": 512, "num_repeats": 4},
+    ],
+}
+CONV_KERNEL_INITIALIZER = {
+    "class_name": "VarianceScaling",
+    "config": {
+        "scale": 2.0,
+        "mode": "fan_out",
+        "distribution": "truncated_normal"
+    }
+}
+
+def Conv2DFixedPadding(filters, kernel_size, strides, name):
+    """Conv2D block with fixed padding.
+    """
+    def apply(inputs):
+        if strides > 1:
+            inputs = fixed_padding(inputs, kernel_size)
+        return layers.Conv2D(
+            filters=filters,
+            kernel_size=kernel_size,
+            strides=strides,
+            padding="same" if strides == 1 else "valid",
+            use_bias=False,
+            kernel_initializer=CONV_KERNEL_INITIALIZER,
+            name=name,
+        )(inputs)
+    return apply
+
+def STEM(bn_momentum: float, bn_epsilon: float, activation: str):
+    """ResNet-D type STEM block."""
+
+    def apply(inputs):
+        bn_axis = 3 if backend.image_data_format() == "channels_last" else 1
+
+        # First stem block
+        x = Conv2DFixedPadding(
+            filters=32, kernel_size=3, strides=2, name="stem_conv_1"
+        )(inputs)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name="stem_batch_norm_1",
+        )(x)
+        x = layers.Activation(activation, name="stem_act_1")(x)
+
+        # Second stem block
+        x = Conv2DFixedPadding(
+            filters=32, kernel_size=3, strides=1, name="stem_conv_2"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name="stem_batch_norm_2",
+        )(x)
+        x = layers.Activation(activation, name="stem_act_2")(x)
+
+        # Final Stem block:
+        x = Conv2DFixedPadding(
+            filters=64, kernel_size=3, strides=1, name="stem_conv_3"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name="stem_batch_norm_3",
+        )(x)
+        x = layers.Activation(activation, name="stem_act_3")(x)
+
+        # Replace stem max pool:
+        x = Conv2DFixedPadding(
+            filters=64, kernel_size=3, strides=2, name="stem_conv_4"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name="stem_batch_norm_4",
+        )(x)
+        x = layers.Activation(activation, name="stem_act_4")(x)
+        return x
+
+    return apply
+
+
+def SE(in_filters: int, se_ratio: float, name: str, expand_ratio: int = 1):
```

Comment:
`name` should go last and default to `None`. You can then use `get_uid` to uniquify the default name if None https://keras.io/api/utils/backend_utils/#getuid-function

Reply:
Done